Baneful Bunker + King's Shield

thechurchofcage opened the pull request:
Both use the same battlescript to apply a move effect from atk49

DizzyEggg commented:
1. For Baneful Bunker you're setting up gHitMarker flags both in a script and in a function.
2. For now ignore kings shield / defiant compatibility, I'll try to think of something later.
3. You could move gMoveResultFlags &= ~(MOVE_RESULT_NO_EFFECT); to the battlescript since you're using it twice in the function. The command is bichalfword
4. You're using move text buffer for abilities in battle_util.c PREPARE_MOVE_BUFFER(gBattleTextBuff1, ABILITY_EFFECT_SPORE);
5. Duplicated     
if (moveType == TYPE_GROUND && defType == TYPE_FLYING && IsBattlerGrounded(battlerDef))
        mod = UQ_4_12(1.0);
6.  What was your motivation behind putting a kings shield in the type effectiveness mulupier getting function?

thechurchofcage commented:
1) Done 👍 
2) What's wrong with the current implementation?
3) Done, thanks 👍 
4) I'm dumb, thanks! Fixed
5) Oops. Fixed
6) That's how King's Shield works now - even if the move would have no effect against the target, it still triggers King's Shield: https://bulbapedia.bulbagarden.net/wiki/King%27s_Shield_(move)#Generation_VII

thechurchofcage commented:
2. Fixed it so that gBattlerAttacker and gBattlerTarget are switched back after King's Shield's effect is applied :P
